environment-api/src/test/java/com/sequenceiq/environment/api/v1/environment/model/request/aws/AWSDiskEncryptionParametersTest.java

```diff
@@ -0,0 +1,101 @@
+    }
+
+    @Test
+    void builderTest() {
+        AWSDiskEncryptionParameters underTest = AWSDiskEncryptionParameters.builder()
+                .withEncryptionKeyArn(ENCRYPTION_KEY_ARN)
+                .build();
+
+        verifyFields(underTest);
+    }
+
+    static Object[][] encryptionKeyArnValidationTestDataProvider() {
+        return new Object[][]{
+                // testCaseName encryptionKeyArn expectedValid
+                {"encryptionKeyArn=null", null, true},
+                {"encryptionKeyArn=ENCRYPTION_KEY_ARN", ENCRYPTION_KEY_ARN, false},
+                {"encryptionKeyArn=arn:aws:kms:us-east-1:012345678910:key/", "arn:aws:kms:us-east-1:012345678910:key/", false},
+                {"encryptionKeyArn=aws:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab", "arn:aws:kms:us-east-1:012345678910:key/\", " +
+                        "\"arn:aws:kms:us-east-1:012345678910:key/\" false},\n" +
+                        "                {\"encryptionKeyArn=aws:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab", false},
+                {"encryptionKeyArn=arn:aws:kms:us-east-1:abc:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws:kms:us-east-1:abc:key/1234abcd-12ab-34cd-56ef-1234567890ab", false},
+                {"encryptionKeyArn=arn:aws:role:us-east-1:012345678910/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws:role:us-east-1:012345678910/1234abcd-12ab-34cd-56ef-1234567890ab", false},
+                {"encryptionKeyArn=arn:aws-us-gov:role:us-east/2:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws-us-gov:role:us-east/2:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab", false},
+                {"encryptionKeyArn=arn:aws-us-gov:role:us-east-2:0123456789-:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws-us-gov:role:us-east-2:0123456789-:key/1234abcd-12ab-34cd-56ef-1234567890ab", false},
+                {"encryptionKeyArn=arn:aws:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab", true},
+                {"encryptionKeyArn=arn:aws-us-gov:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws-us-gov:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab", true},
+                {"encryptionKeyArn=arn:aws-cn:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws-cn:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab", true},
+                {"encryptionKeyArn=arn:aws-us-gov:kms:us-east-2:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws-us-gov:kms:us-east-2:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab", true},
+
```

Comment:
Perhaps add one or two more cases for the empty region & account ID scenario (both of which should result in a validation error).